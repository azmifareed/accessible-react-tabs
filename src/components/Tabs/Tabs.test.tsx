import React from 'react';
import { render, screen, fireEvent } from '../../utils/test-utils';
import { Tabs, Tab } from '.';

function renderTabs() {
  return render(<Tabs ariaLabel="test label">
    <Tab id="tab1" label="Tab 1" isSelected={true}>tab panel 1</Tab>
    <Tab id="tab2" label="Tab 2">tab panel 2</Tab>
  </Tabs>)
}

test('renders Tabs correctly', () => {
  const component = renderTabs();
  expect(component.baseElement).toMatchSnapshot();
});

test('handles tab nav item click correctly', () => {
  const component = renderTabs();
  let tab;

  fireEvent.click(component.getByText(/Tab 2/i));
  tab = screen.getByText(/Tab 2/i);
  expect(tab.attributes.getNamedItem('aria-selected')).toBeTruthy;
  tab = screen.getByText(/Tab 1/i);
  expect(tab.attributes.getNamedItem('aria-selected')).toBeFalsy;
  expect(component.baseElement).toMatchSnapshot();

  fireEvent.click(component.getByText(/Tab 1/i));
  tab = screen.getByText(/Tab 1/i);
  expect(tab.attributes.getNamedItem('aria-selected')).toBeTruthy;
  tab = screen.getByText(/Tab 2/i);
  expect(tab.attributes.getNamedItem('aria-selected')).toBeFalsy;
  expect(component.baseElement).toMatchSnapshot();
});

test('handles tab nav item left arrow keypress correctly', () => {
  const component = renderTabs();

  fireEvent.keyDown(component.getByText(/Tab 1/i),{ key: 'ArrowLeft', code: 'ArrowLeft', which: 37, keyCode: 37});
  expect(component.baseElement).toMatchSnapshot();


  fireEvent.keyDown(component.getByText(/Tab 2/i),{ key: 'ArrowLeft', code: 'ArrowLeft', which: 37, keyCode: 37});
  expect(component.baseElement).toMatchSnapshot();
});

test('handles tab nav item right arrow keypress correctly', () => {
  const component = renderTabs();

  fireEvent.keyDown(component.getByText(/Tab 1/i),{ key: 'ArrowRight', code: 'ArrowRight', which: 39, keyCode: 39});
  expect(component.baseElement).toMatchSnapshot();


  fireEvent.keyDown(component.getByText(/Tab 2/i),{ key: 'ArrowRight', code: 'ArrowRight', which: 39, keyCode: 39});
  expect(component.baseElement).toMatchSnapshot();
});



test('handles tab nav item arrow keypress correctly when no item is selected', () => {
const component = render(<Tabs ariaLabel="test label">
    <Tab id="tab1" label="Tab 1">tab panel 1</Tab>
    <Tab id="tab2" label="Tab 2">tab panel 2</Tab>
  </Tabs>)

  fireEvent.keyDown(component.getByText(/Tab 1/i),{ key: 'ArrowLeft', code: 'ArrowLeft', which: 37, keyCode: 37});
  expect(component.baseElement).toMatchSnapshot();


  fireEvent.keyDown(component.getByText(/Tab 2/i),{ key: 'ArrowRight', code: 'ArrowRight', which: 39, keyCode: 39});
  expect(component.baseElement).toMatchSnapshot();
});

