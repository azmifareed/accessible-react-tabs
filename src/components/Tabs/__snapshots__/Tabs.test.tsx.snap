// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`handles tab nav item arrow keypress correctly when no item is selected 1`] = `
<body>
  <div>
    <div
      class="sc-hKFxyN bPCfgy"
    >
      <div
        aria-label="test label"
        class="sc-bdnxRM ewiLTn"
        role="tablist"
      >
        <button
          aria-controls="tab1-tab"
          aria-selected="false"
          class="sc-gtsrHT crbYRc"
          id="tab1"
          role="tab"
          tabindex="-1"
        >
          Tab 1
        </button>
        <button
          aria-controls="tab2-tab"
          aria-selected="true"
          class="sc-gtsrHT hLdcqK"
          id="tab2"
          role="tab"
        >
          Tab 2
        </button>
      </div>
      <div
        aria-labelledby="tab1"
        class="sc-dlnjwi dBvEUx"
        id="tab1-tab"
        role="tabpanel"
        tabindex="0"
      >
        tab panel 1
      </div>
      <div
        aria-labelledby="tab2"
        class="sc-dlnjwi cIarac"
        id="tab2-tab"
        role="tabpanel"
        tabindex="0"
      >
        tab panel 2
      </div>
    </div>
  </div>
</body>
`;

exports[`handles tab nav item arrow keypress correctly when no item is selected 2`] = `
<body>
  <div>
    <div
      class="sc-hKFxyN bPCfgy"
    >
      <div
        aria-label="test label"
        class="sc-bdnxRM ewiLTn"
        role="tablist"
      >
        <button
          aria-controls="tab1-tab"
          aria-selected="true"
          class="sc-gtsrHT hLdcqK"
          id="tab1"
          role="tab"
        >
          Tab 1
        </button>
        <button
          aria-controls="tab2-tab"
          aria-selected="false"
          class="sc-gtsrHT crbYRc"
          id="tab2"
          role="tab"
          tabindex="-1"
        >
          Tab 2
        </button>
      </div>
      <div
        aria-labelledby="tab1"
        class="sc-dlnjwi cIarac"
        id="tab1-tab"
        role="tabpanel"
        tabindex="0"
      >
        tab panel 1
      </div>
      <div
        aria-labelledby="tab2"
        class="sc-dlnjwi dBvEUx"
        id="tab2-tab"
        role="tabpanel"
        tabindex="0"
      >
        tab panel 2
      </div>
    </div>
  </div>
</body>
`;

exports[`handles tab nav item click correctly 1`] = `
<body>
  <div>
    <div
      class="sc-hKFxyN bPCfgy"
    >
      <div
        aria-label="test label"
        class="sc-bdnxRM ewiLTn"
        role="tablist"
      >
        <button
          aria-controls="tab1-tab"
          aria-selected="false"
          class="sc-gtsrHT crbYRc"
          id="tab1"
          role="tab"
          tabindex="-1"
        >
          Tab 1
        </button>
        <button
          aria-controls="tab2-tab"
          aria-selected="true"
          class="sc-gtsrHT hLdcqK"
          id="tab2"
          role="tab"
        >
          Tab 2
        </button>
      </div>
      <div
        aria-labelledby="tab1"
        class="sc-dlnjwi dBvEUx"
        id="tab1-tab"
        role="tabpanel"
        tabindex="0"
      >
        tab panel 1
      </div>
      <div
        aria-labelledby="tab2"
        class="sc-dlnjwi cIarac"
        id="tab2-tab"
        role="tabpanel"
        tabindex="0"
      >
        tab panel 2
      </div>
    </div>
  </div>
</body>
`;

exports[`handles tab nav item click correctly 2`] = `
<body>
  <div>
    <div
      class="sc-hKFxyN bPCfgy"
    >
      <div
        aria-label="test label"
        class="sc-bdnxRM ewiLTn"
        role="tablist"
      >
        <button
          aria-controls="tab1-tab"
          aria-selected="true"
          class="sc-gtsrHT hLdcqK"
          id="tab1"
          role="tab"
        >
          Tab 1
        </button>
        <button
          aria-controls="tab2-tab"
          aria-selected="false"
          class="sc-gtsrHT crbYRc"
          id="tab2"
          role="tab"
          tabindex="-1"
        >
          Tab 2
        </button>
      </div>
      <div
        aria-labelledby="tab1"
        class="sc-dlnjwi cIarac"
        id="tab1-tab"
        role="tabpanel"
        tabindex="0"
      >
        tab panel 1
      </div>
      <div
        aria-labelledby="tab2"
        class="sc-dlnjwi dBvEUx"
        id="tab2-tab"
        role="tabpanel"
        tabindex="0"
      >
        tab panel 2
      </div>
    </div>
  </div>
</body>
`;

exports[`handles tab nav item left arrow keypress correctly 1`] = `
<body>
  <div>
    <div
      class="sc-hKFxyN bPCfgy"
    >
      <div
        aria-label="test label"
        class="sc-bdnxRM ewiLTn"
        role="tablist"
      >
        <button
          aria-controls="tab1-tab"
          aria-selected="false"
          class="sc-gtsrHT crbYRc"
          id="tab1"
          role="tab"
          tabindex="-1"
        >
          Tab 1
        </button>
        <button
          aria-controls="tab2-tab"
          aria-selected="true"
          class="sc-gtsrHT hLdcqK"
          id="tab2"
          role="tab"
        >
          Tab 2
        </button>
      </div>
      <div
        aria-labelledby="tab1"
        class="sc-dlnjwi dBvEUx"
        id="tab1-tab"
        role="tabpanel"
        tabindex="0"
      >
        tab panel 1
      </div>
      <div
        aria-labelledby="tab2"
        class="sc-dlnjwi cIarac"
        id="tab2-tab"
        role="tabpanel"
        tabindex="0"
      >
        tab panel 2
      </div>
    </div>
  </div>
</body>
`;

exports[`handles tab nav item left arrow keypress correctly 2`] = `
<body>
  <div>
    <div
      class="sc-hKFxyN bPCfgy"
    >
      <div
        aria-label="test label"
        class="sc-bdnxRM ewiLTn"
        role="tablist"
      >
        <button
          aria-controls="tab1-tab"
          aria-selected="true"
          class="sc-gtsrHT hLdcqK"
          id="tab1"
          role="tab"
        >
          Tab 1
        </button>
        <button
          aria-controls="tab2-tab"
          aria-selected="false"
          class="sc-gtsrHT crbYRc"
          id="tab2"
          role="tab"
          tabindex="-1"
        >
          Tab 2
        </button>
      </div>
      <div
        aria-labelledby="tab1"
        class="sc-dlnjwi cIarac"
        id="tab1-tab"
        role="tabpanel"
        tabindex="0"
      >
        tab panel 1
      </div>
      <div
        aria-labelledby="tab2"
        class="sc-dlnjwi dBvEUx"
        id="tab2-tab"
        role="tabpanel"
        tabindex="0"
      >
        tab panel 2
      </div>
    </div>
  </div>
</body>
`;

exports[`handles tab nav item right arrow keypress correctly 1`] = `
<body>
  <div>
    <div
      class="sc-hKFxyN bPCfgy"
    >
      <div
        aria-label="test label"
        class="sc-bdnxRM ewiLTn"
        role="tablist"
      >
        <button
          aria-controls="tab1-tab"
          aria-selected="false"
          class="sc-gtsrHT crbYRc"
          id="tab1"
          role="tab"
          tabindex="-1"
        >
          Tab 1
        </button>
        <button
          aria-controls="tab2-tab"
          aria-selected="true"
          class="sc-gtsrHT hLdcqK"
          id="tab2"
          role="tab"
        >
          Tab 2
        </button>
      </div>
      <div
        aria-labelledby="tab1"
        class="sc-dlnjwi dBvEUx"
        id="tab1-tab"
        role="tabpanel"
        tabindex="0"
      >
        tab panel 1
      </div>
      <div
        aria-labelledby="tab2"
        class="sc-dlnjwi cIarac"
        id="tab2-tab"
        role="tabpanel"
        tabindex="0"
      >
        tab panel 2
      </div>
    </div>
  </div>
</body>
`;

exports[`handles tab nav item right arrow keypress correctly 2`] = `
<body>
  <div>
    <div
      class="sc-hKFxyN bPCfgy"
    >
      <div
        aria-label="test label"
        class="sc-bdnxRM ewiLTn"
        role="tablist"
      >
        <button
          aria-controls="tab1-tab"
          aria-selected="true"
          class="sc-gtsrHT hLdcqK"
          id="tab1"
          role="tab"
        >
          Tab 1
        </button>
        <button
          aria-controls="tab2-tab"
          aria-selected="false"
          class="sc-gtsrHT crbYRc"
          id="tab2"
          role="tab"
          tabindex="-1"
        >
          Tab 2
        </button>
      </div>
      <div
        aria-labelledby="tab1"
        class="sc-dlnjwi cIarac"
        id="tab1-tab"
        role="tabpanel"
        tabindex="0"
      >
        tab panel 1
      </div>
      <div
        aria-labelledby="tab2"
        class="sc-dlnjwi dBvEUx"
        id="tab2-tab"
        role="tabpanel"
        tabindex="0"
      >
        tab panel 2
      </div>
    </div>
  </div>
</body>
`;

exports[`renders Tabs correctly 1`] = `
<body>
  <div>
    <div
      class="sc-hKFxyN bPCfgy"
    >
      <div
        aria-label="test label"
        class="sc-bdnxRM ewiLTn"
        role="tablist"
      >
        <button
          aria-controls="tab1-tab"
          aria-selected="true"
          class="sc-gtsrHT hLdcqK"
          id="tab1"
          role="tab"
        >
          Tab 1
        </button>
        <button
          aria-controls="tab2-tab"
          aria-selected="false"
          class="sc-gtsrHT crbYRc"
          id="tab2"
          role="tab"
          tabindex="-1"
        >
          Tab 2
        </button>
      </div>
      <div
        aria-labelledby="tab1"
        class="sc-dlnjwi cIarac"
        id="tab1-tab"
        role="tabpanel"
        tabindex="0"
      >
        tab panel 1
      </div>
      <div
        aria-labelledby="tab2"
        class="sc-dlnjwi dBvEUx"
        id="tab2-tab"
        role="tabpanel"
        tabindex="0"
      >
        tab panel 2
      </div>
    </div>
  </div>
</body>
`;
